app/src/main/java/de/westnordost/streetcomplete/quests/diet_type/AddVegetarian.kt

```diff
@@ -31,5 +31,7 @@ class AddVegetarian : OsmFilterQuestType<DietAvailability>() {
 
     override fun applyAnswerTo(answer: DietAvailability, changes: StringMapChangesBuilder) {
         changes.updateWithCheckDate("diet:vegetarian", answer.osmValue)
+        if (answer.osmValue == "no")
+            changes.deleteIfExists("diet:vegan")
```

Comment:
I don't know SC exact implementation, but would not this also delete `diet:vegan = no`?

Reply:
yes

Reply:
But is not this problematic as it removes correct information (or actually more detailed information in case of `only`)?

Reply:
Can you point me to a restaurant that doesn't have vegetarian food but only has vegan food?

Reply:
1) In case that it is "no" it is just not needed duplicate

2) in case that it is not no it would be wrong (or SC mapper is mistaken)

Reply:
Initial idea of this PR was to remove vegan=yes only if SC mapper answered vegetarian=no, because
* the SC user could not have tagged vegan before getting the vegetarian quest (it's simply not asked)
* assume that SC user does give a correct answer (vegan=yes might be outdated)

I decided to remove vegan tag always if vegetarian=no because it's easier and the tag is not needed as its value must be no.

Reply:
Hmm yeah, it all seems to depend on the fact that the user's answer is correct. I guess that's a fair thing to assume, respectively we cannot handle it otherwise.

Reply:
Well yes, because if we assume that users will give incorrect answers, we might as well disable all quests altogether, not just this one :smile:

Reply:
>Can you point me to a restaurant that doesn't have vegetarian food but only has vegan food?
@westnordost not sure if it's still relevant, but as nobody answered I will :).

There are growing amounts, but probably less than 100 in Germany. Two examples: 
https://www.xn--tomundmx-6za.de/ and http://www.die-gartenkueche.com/ (ate a cake there once, was nice!)

Reply:
![image](https://user-images.githubusercontent.com/4661658/134050258-d868b487-b87b-43b5-95ad-47668a58bc8c.png)

Daal-soup is vegetarian. And vegan.

Reply:
No, at the bottom right it says "alles vegan" :)

Reply:
At [happycow](https://www.happycow.net/searchmap/?s=3&location=munich&veganStores=on&filters=vegan-bakery-bnb-catering-coffee-delivery-farmers-foodtruck-health-icecream-juicebar-marketvendor-organization-other-veganprofessional-spa-vegshop&metric=km&limit=81&order=default&lat=48.134255858400735&lng=11.580030835505575&zoom=15.25&page=1&bb=48.12187655432498%2C11.56270890312349%2C48.146635162476485%2C11.597352767887658) you can also search for all-vegan stores, restaurants and co (does not use OSM as data source though).

Reply:
@carstenhag I though the question _"name a vegan restaurant which is not **also** a vegetarian restaurant"_ was supposed to be rhetoric. As far as I know, `vegan` is subset of `vegetarian`, so by definition, if it is `vegan`, it also **must** be `vegetarian`. (Same for example as every _electric car_ is also **always** a _motor vehicle_)

Reply:
Not sure if I got you, but yeah, vegan is the most strict, vegetarian is less strict.  
But the tag can also have a diet:vegan=only value, then diet:vegetarian=yes should imo not be set. 

(This was just to try to explain again, I am not sure if it's implemented like this already)

Reply:
vegetarian is not asked is a place is vegan only